src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int AMOUNT_OF_FIGURE = 5;
+    private static final double MAX_PARAMETER = 10.0;
+    private static final double DEFAULT_RADIUS = 10.0;
+
+    public Figure getRandomFigure() {
+        switch (new Random().nextInt(AMOUNT_OF_FIGURE)) {
+            case 0:
+                return new Square(Color.values()[new Random().nextInt(Color.values().length)],
```

Comment:
you have this logic in your ColorSupplier - let's use it instead
create only 1 object of ColorSupplier inside method